playbooks/setup_nodepool.yml

```diff
@@ -0,0 +1,244 @@
+---
+# We need to target all hosts in the group first to gather facts,
+# otherwise any tasks using the group host_vars will fail.
+- name: Gather facts from nodepool_server group
+  hosts: "nodepool_server"
+  gather_facts: no
+  user: root
+  tasks:
+    - name: Wait for a successful connection
+      wait_for_connection:
+        connect_timeout: 2
+        delay: 5
+        sleep: 5
+        timeout: 60
+
+    - name: Gather facts
+      setup:
+        gather_subset: "!facter,!ohai"
+
+- name: Setup nodepool servers serially
+  hosts: "nodepool_server"
+  gather_facts: no
+  user: root
+  # NOTE(odyssey4me):
+  # The zookeeper service loses all its data when it restarts,
+  # which may happen when packages are upgraded. To ensure that
+  # this is not an issue for us we execute the playbook in
+  # serial so that the cluster always retains its quorum.
+  serial: 1
+  vars_files:
+    - vars/nodepool.yml
+  handlers:
+    - name: Restart zookeeper
+      service:
+        name: zookeeper
+        daemon_reload: yes
+        state: restarted
+      listen: Restart all services
+
+    - name: Restart nodepool-builder
+      service:
+        name: nodepool-builder
+        daemon_reload: yes
+        state: restarted
+      listen: Restart all services
+
+    - name: Restart nodepool-launcher
+      service:
+        name: nodepool-launcher
+        daemon_reload: yes
+        state: restarted
+      listen: Restart all services
+
+  tasks:
+    - name: Wait for a successful connection
+      wait_for_connection:
+        connect_timeout: 2
+        delay: 5
+        sleep: 5
+        timeout: 60
+
+    - name: Gather facts
```

Comment:
Do you need the `Gather facts` and `Wait for a successful connection` tasks here since they're both run above?

Reply:
Done.